lib/datadog/appsec/contrib/graphql/integration.rb

```diff
@@ -24,13 +31,19 @@ def self.loaded?
           end
 
           def self.compatible?
-            super && version >= MINIMUM_VERSION
+            super && version >= MINIMUM_VERSION && ast_node_classes_defined?
           end
 
           def self.auto_instrument?
             true
           end
 
+          def self.ast_node_classes_defined?
```

Comment:
Adding a couple of test cases 

1. Test positive with our graphql appraisal groups.
2. Test negative with `hide_const`

Reply:
added here: [9522f70](https://github.com/DataDog/dd-trace-rb/pull/3887/commits/9522f70bd948866ef9de043a4b9ed10a3a32aee1)